core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -197,7 +198,7 @@ protected String targetBranch() {
   }
 
   protected ExecutorService workerPool() {
-    return this.workerPool;
+    return Optional.ofNullable(workerPool).orElseGet(ThreadPools::getWorkerPool);
```

Comment:
I have checked the code, and there are some places where the `workerPool` is accessed directly.
Is this a fancy way of telling:
```
if (workerPool == null) {
    this.workerPool = ThreadPools.getWorkerPool();
}

return workerPool;
```

Reply:
yeah, I think it's a cool one-liner, assuming that repeatedly calling ThreadPools::getWorkerPool is not expensive, which is true because it just returns the pool from the 2nd call:
```
  public static ExecutorService getWorkerPool() {
    return WORKER_POOL;
  }
```

Reply:
btw, looks like I forgot to replace with workerPool(), let me do quickly

Reply:
If you don't mind, I would still prefer the first version where we call the getWorkerPool only once.

Reply:
@abstractdog you should use `orElseGet` - it's lazy

Reply:
`orElseGet` is lazy

Reply:
Thx!